src/components/ItemsList.js

```diff
@@ -1,32 +1,44 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import firebase from '../lib/firebase';
-import { useCollectionData } from 'react-firebase-hooks/firestore';
+
+const userToken = localStorage.getItem('token');
+
+const db = firebase.firestore().collection('shopping_list');
 
 const ItemsList = () => {
-  const [itemsList, loading, error] = useCollectionData(
-    firebase.firestore().collection('items'),
-    {
-      snapshotListenOptions: { includeMetadataChanges: true },
-    },
-  );
-  // useCollectionData extracts firebase.firestore.QuerySnapshot.docs while useCollection
-  // returns firebase.firestore.QuerySnapshot itself.
+  const [items, setItems] = useState('');
+
+  useEffect(() => {
+    let mounted = true;
+    db.where('token', '==', userToken)
+      .get()
+      .then((data) => {
+        if (data.docs.length) {
+          let itemsList = data.docs[0]
+            .data()
+            .items.map((element) => element.shoppingListItemName);
+          if (mounted) {
+            setItems(itemsList);
+          }
+        }
+      });
+    return () => (mounted = false);
+  });
```

Comment:
This code may be causing the quota issue. It runs in a loop

Reply:
does giving [items] as the second argument of the useEffect help ?

Reply:
..or just remove the useEffect completely and the call to the firestore happens only once :)